.github/workflows/pr-preview.yml

```diff
@@ -75,17 +79,28 @@ jobs:
           if [ "${BRANCH}" = "release/demo" ]
           then
             aws s3 cp "${WHEELFILE}" s3://core-previews/${PREVIEW_BRANCH}/streamlit-11.11.11-py2.py3-none-any.whl --acl public-read
-            echo -e "Wheel file download link: https://core-previews.s3-us-west-2.amazonaws.com/${PREVIEW_BRANCH}/streamlit-11.11.11-py2.py3-none-any.whl"
+            S3_URL="https://core-previews.s3-us-west-2.amazonaws.com/${PREVIEW_BRANCH}/streamlit-11.11.11-py2.py3-none-any.whl"
           else
             aws s3 cp "${WHEELFILE}" s3://core-previews/${PREVIEW_BRANCH}/ --acl public-read
-            echo -e "Wheel file download link: https://core-previews.s3-us-west-2.amazonaws.com/${PREVIEW_BRANCH}/${WHEELFILE}"
+            S3_URL="https://core-previews.s3-us-west-2.amazonaws.com/${PREVIEW_BRANCH}/${WHEELFILE}"
           fi
+          echo -e "Wheel file download link: ${S3_URL}"
 
           cd ../..
           # env variables don't carry over between gh action jobs
           echo "enable-setup=${{ env.AWS_ACCESS_KEY_ID != '' }}" >> $GITHUB_OUTPUT
           echo "preview-branch=$PREVIEW_BRANCH" >> $GITHUB_OUTPUT
-          echo "s3-url=https://core-previews.s3-us-west-2.amazonaws.com/${PREVIEW_BRANCH}/${WHEELFILE}" >> $GITHUB_OUTPUT
+          echo "s3-url=${S3_URL}" >> $GITHUB_OUTPUT
+
+      - name: Set success deployment status
+        if: success() && github.repository == 'streamlit/streamlit' && github.event_name == 'pull_request' && github.event.sender.type == 'User'
+        uses: pbrandone/create-status-action@b1112f45e23466b401db7d9fc1aa349fb84eacfc
```

Comment:
Just for my understanding - what are the other options for `github.event.sender.type` outside of 'User'? Like would a dependabot triggered PR be something else? 
Did some searching and didn't see a clear explanation of this field in the docs, but would appreciate a link if you have one you used!

Reply:
When Dependabot creates a PR, a read-only token is used, so this operation always fails.

> For workflows initiated by Dependabot (github.actor == 'dependabot[bot]') using the pull_request, pull_request_review, pull_request_review_comment, push, create, deployment, and deployment_status events, the following restrictions apply:
> 
> * GITHUB_TOKEN has read-only permissions by default.
> * Secrets are populated from Dependabot secrets. GitHub Actions secrets are not available.

https://docs.github.com/en/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions

Reply:
Sorry let me rephrase - is there ever a case where `github.event.sender.type` is not 'User' ? 
Essentially just curious when is `github.event.sender.type` something other than 'User'.

Reply:
When PR is created by `dependabot`, this field is not set to `User` value and we must have this check because `dependabot` has limited token.